src/engraving/libmscore/measure.cpp

```diff
@@ -4115,12 +4077,77 @@ static bool hasAccidental(Segment* s)
 }
 
 //---------------------------------------------------------
-//   computeMinWidth
-//    sets the minimum stretched width of segment list s
-//    set the width and x position for all segments
+//      minSysTicks
+//      returns the shortest note/rest in the system
+//---------------------------------------------------------
+
+Fraction Measure::minSysTicks()
+{
+    Fraction minTicks = Fraction(10, 1);
+    for (MeasureBase* mb : system()->measures()) {
+        if (mb->isMeasure()) {
+            Measure* m = toMeasure(mb);
+            Fraction curMinTicks = m->computeTicks();
+            if (curMinTicks < minTicks) {
+                minTicks = curMinTicks;
+            }
+        }
+    }
+    return minTicks;
+}
+
+//---------------------------------------------------------
+//      Stretch formula
+//      Computes the stretch for a given segment depending
+//      on its duration with respect to the shortest one.
+//      Three different options proposed (see documentation).
+//---------------------------------------------------------
+qreal Measure::stretchFormula(Fraction curTicks, Fraction minTicks, qreal stretchCoeff)
+{
+    qreal slope = userStretch() * score()->styleD(Sid::measureSpacing);
+
+    if (curTicks > m_timesig) { // This is the case of MM rests
+        curTicks = curTicks - m_timesig; // 2 bars MM rests is spaced same as 1 bar
+        if (curTicks > m_timesig * 20) { // To avoid excessive
+            curTicks = m_timesig * 20;
+        }
+    }
+
+    if (minTicks <= Fraction(1, 32)) {
+        // Reduces the slope of the spacing curve in case very short notes are present.
+        // Avoids having longer notes too wide.
+        qreal reduction = qMax((1 - 0.2 * log2(qreal(Fraction(1, 16).ticks()) / qreal(minTicks.ticks()))), 0.3);
+        slope = slope * reduction;
+    }
+
+    // Linear spacing
+    //qreal str = 1 - 0.112 * slope +  0.112 * slope * qreal(curTicks.ticks()) / qreal(minTicks.ticks());
+    // Logarithmic spacing (MS 3.6)
+    //qreal str = 1.0 + 0.721 * slope * log(qreal(curTicks.ticks()) / qreal(minTicks.ticks()));
+    // Quadratic spacing (Gould)
+    qreal str = 1 - 0.647 * slope + 0.647 * slope * sqrt(qreal(curTicks.ticks()) / qreal(minTicks.ticks()));
```

Comment:
avoid the usage of 'magic numbers'. I suggest to declare a named variable here or at the top of this file, like:

static constexpr float STRETCH_FACTOR = 0.647;